examples/js/loaders/GLTFLoader.js

```diff
@@ -1547,59 +1528,64 @@ THREE.GLTFLoader = ( function () {
 
 	}
 
-	function isArrayEqual( a, b ) {
+	function createGeometryKey( primitiveDef ) {
 
-		if ( a.length !== b.length ) return false;
+		var dracoExtension = primitiveDef.extensions && primitiveDef.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ];
+		var geometryKey;
 
-		for ( var i = 0, il = a.length; i < il; i ++ ) {
+		if ( dracoExtension ) {
 
-			if ( a[ i ] !== b[ i ] ) return false;
+			geometryKey = 'draco:' + dracoExtension.bufferView
+				+ ':' + dracoExtension.indices
+				+ ':' + createAttributesKey( dracoExtension.attributes );
+
+		} else {
+
+			geometryKey = primitiveDef.indices + ':' + createAttributesKey( primitiveDef.attributes ) + primitiveDef.mode;
 
 		}
 
-		return true;
+		return geometryKey;
 
 	}
 
-	function getCachedGeometry( cache, newPrimitive ) {
+	function createAttributesKey( attributes ) {
 
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var attributesKey = '';
 
-			var cached = cache[ i ];
+		for ( var key in attributes ) {
 
-			if ( isPrimitiveEqual( cached.primitive, newPrimitive ) ) return cached.promise;
+			attributesKey += key + attributes[ key ];
 
 		}
 
-		return null;
+		return attributesKey;
 
 	}
 
-	function getCachedCombinedGeometry( cache, geometries ) {
+	function createArrayKey( a ) {
 
```

Comment:
So I assume I can call `createAttributesKey` over the elements of the array?

Reply:
As written here, the parameter to `createArrayKey` will be one of:

- Array<THREE.BufferGeometry>
- Array<GLTF.Primitive>

I would just split this into two methods, or do this in a `for` loop  inline (since each method would only be called once). For each geometry you can hash on `geometry.uuid`. For each primitive you'll need to use the `createGeometryKey` method above. The mesh/primitive and mesh/geometry concepts in glTF and three.js are a little different, which makes naming these things hard... 😅